### ISSUE TITLE
Code snippets not showing (Uncaught SyntaxError: Unexpected token '.' (at book.js:1:1))

### ISSUE THREAD

**Issue by antyj1994** [google/comprehensive-rust#733]
![image](https://github.com/google/comprehensive-rust/assets/32547268/742fb3e0-d6d0-4274-bffc-35d41f1a1226)

Just started this course, error showing when running comprehensive rust in local environment.

Code snippets appear for a fraction of second and then disappear with this error in console.

**Comment by mgeisler** [comment]
Hi @antyj1994, thanks for reporting this!

I'm not able to reproduce it. When I run `mdbook serve`, and open http://localhost:3000/, I see the code snippets like normal. What do you see when you browse https://google.github.io/comprehensive-rust/?

How do you build the course locally, with `mdbook build` or `mdbook serve`?

Are you on Linux or Windows or...? I've only tested the book on Linux.

I just tried reinstalling `mdbook` version 0.4.30 with `cargo install mdbook` and it works the same.

**Comment by mgeisler** [comment]
I [tweeted about this](https://twitter.com/mrtngslr/status/1663834836530786307) — let's see if someone else can help us debug this.

**Comment by mauvealerts** [comment]
I had this happen on Windows, not my Linux machine. I took a look and `http://localhost:3000/book.js` contains `../third_party/mdbook/book.js`, which made me think of symlink issues. Indeed `themes/book.js` is supposed to be a symlink.

Apparently with `git config --global core.symlinks true` symlinks are created, but only if it's possible to do so without UAC prompt (via enabling developer mode). I didn't notice a warning about the fallback, which is unfortunate. I just deleted my repo and re-cloned after fixing my settings.

**Comment by drhcc** [comment]
@mauvealerts 

> and re-cloned after fixing my settings. 

would you please give more details on what settings you have fixed and how you've done it? Thank you

**Comment by mauvealerts** [comment]
The settings are:
* Enable Git symlinks via `git config --global core.symlinks true`
* Enable Developer Mode (on my Windows 11 machine, it's under Privacy & security > For developers)

I did some more testing and the can be fixed via `git restore`, `git reset --hard`, etc. (instead of deleting and re-cloning).

**Comment by antyj1994** [comment]
Thanks for solving this issue. 
From my side enabling symlinks via `git config --global core.symlinks true` then `git restore` and `git reset --hard` did not work. 
I had to delete and clone the repo again but now it's working fine.

**Comment by djmitche** [comment]
If one of you could create a PR to update the README with a note about this, that would be great! Maybe just suggest `git config --global core.symlinks true` and include a link to this issue for more information?

**Comment by mauvealerts** [comment]
#760 added a brief note to the README.

**Comment by fphartnc** [comment]
I had a similar issue on Windows 10 -- no code display at all in the snippets, just line numbers.  I also downloaded the .zip file and did not do a git clone; I used 7-zip to unzip and extract the directory.  For those who might not wish to put more write wear on their SSDs than is absolutely necessary, the symlink in Windows can be created using just a Command Prompt with Administrator privileges.  After opening such a command prompt then `cd C:\Users\<username>\comprehensive-rust-main\theme` and rename the existing book.js file to some other name.  Then use the following to create the symlink in Windows:  `mklink book.js ..\third_party\mdbook\book.js`  Note the use of the backslashes here compared to *nix usage.

After the command completes, a dir command will show that a SYMLINK has been created for book.js and the target of the symlink will appear to the right in brackets.  For those who may be corporate environments where Developer mode is disabled by policy, this method has the advantage of not needing Developer mode enabled.

Everything appears to work now... what I don't know is whether this was the only broken symlink... there might be others!

**Comment by mauvealerts** [comment]
> Thanks for solving this issue. From my side enabling developer mode and symlinks via `git config --global core.symlinks true` then `git restore` and `git reset --hard` did not work. I had to delete and clone the repo again but now it's working fine.

The problem is that, if you didn't have symlinks enabled before the clone, `git clone` writes a `.git/config` with `symlinks = false`  which overrides the global setting. So, to "repair" a repo after cloning you need to  `git config core.symlinks true`  inside the repo then `git restore some/path` or `git reset --hard`

> Everything appears to work now... what I don't know is whether this was the only broken symlink... there might be others!

At the time of this writing there are two symlinks in the repo:

* `src/android/interoperability/cpp/overview.svg` → `third_party/cxx/overview.svg`
* `theme/book.js`  → `third_party/mdbook/book.js`

The above list will become out of date over time, of course. You can enumerate the symlinks by running (in PowerShell) `git ls-files --format='%(objectmode) %(path)' | Select-String '^120000 '`

**Comment by fphartnc** [comment]
@mauvalerts: Thanks for pointing out the other symlink.  I was able to fix that using a process similar to my previous post.  I do wish to emphasize here that I pulled the .zip file from the repository and did not do a git clone, and it appears that the .zip format might suffer from this problem of busted symlinks indefinitely.  If so, the developers may wish to consider deprecating availability of the .zip format.   For myself, I have no problem with going over to git clone (I have git-bash installed on Windows already), but others might be impacted.

**Comment by mauvealerts** [comment]
Using GitHub's "Download ZIP" is unlikely to work well on Windows. Although the symlinks are stored with correct metadata, the couple utilities I've tried either:
* Don't handle symlinks at all
* Don't translate the path (i.e. replace `/` with `\`), resulting in broken links

I don't know of a way to disable GitHub's "Download ZIP" option, but none of the material links to it either (AFAICT)... There's a link to a zip file with the exercises, but that doesn't contain symlinks.

**Comment by mgeisler** [comment]
> I don't know of a way to disable GitHub's "Download ZIP" option, but none of the material links to it either (AFAICT)...

Yeah, this is not something we can or should disable — it's outside our area  so to speak 🙂. The note in the README about symlinks should help here. If GitHub has more documentation about this, then I would be happy to point to that too.

For now, I think we can close this issue since there is a documented workaround and since the problem is not specific to the course.
